presto-main/src/test/java/com/facebook/presto/execution/TestSqlTaskExecution.java

```diff
@@ -0,0 +1,1338 @@
+            TaskOutputOperatorFactory taskOutputOperatorFactory = new TaskOutputOperatorFactory(
+                    1,
+                    TABLE_SCAN_NODE_ID,
+                    outputBuffer,
+                    Function.identity(),
+                    new PagesSerdeFactory(new TestingBlockEncodingSerde(new TestingTypeManager()), false));
+            LocalExecutionPlan localExecutionPlan = new LocalExecutionPlan(
+                    ImmutableList.of(new DriverFactory(
+                            0,
+                            true,
+                            true,
+                            ImmutableList.of(testingScanOperatorFactory, taskOutputOperatorFactory),
+                            OptionalInt.empty(),
+                            executionFlow)),
+                    ImmutableList.of(TABLE_SCAN_NODE_ID));
+            TaskContext taskContext = newTestingTaskContext(taskNotificationExecutor, driverYieldExecutor, taskStateMachine);
+            SqlTaskExecution sqlTaskExecution = SqlTaskExecution.createSqlTaskExecution(
+                    taskStateMachine,
+                    taskContext,
+                    outputBuffer,
+                    ImmutableList.of(),
+                    localExecutionPlan,
+                    taskExecutor,
+                    taskNotificationExecutor,
+                    new QueryMonitor(new ObjectMapperProvider().get(), jsonCodec(StageInfo.class), new EventListenerManager(), new NodeInfo("test"), new NodeVersion("testVersion"), new QueryMonitorConfig()));
+
+            //
+            // test body
+            assertEquals(taskStateMachine.getState(), TaskState.RUNNING);
+
+            switch (executionFlow) {
+                case UNGROUPED_EXECUTION:
+                    // add source for pipeline
+                    sqlTaskExecution.addSources(ImmutableList.of(new TaskSource(
+                            TABLE_SCAN_NODE_ID,
+                            ImmutableSet.of(newScheduledSplit(0, TABLE_SCAN_NODE_ID, DriverLifeCycle.taskWide(), 100000, 100123)),
+                            false)));
+                    // assert that partial task result is produced
+                    outputBufferConsumer.consume(123, ASSERT_WAIT_TIMEOUT);
+
+                    // pause operator execution to make sure that
+                    // * operatorFactory will be closed even though operator can't execute
+                    // * completedDriverGroups will NOT include the newly scheduled driver group while pause is in place
+                    testingScanOperatorFactory.getPauser().pause();
+                    // add source for pipeline, mark as no more splits
+                    sqlTaskExecution.addSources(ImmutableList.of(new TaskSource(
+                            TABLE_SCAN_NODE_ID,
+                            ImmutableSet.of(
+                                    newScheduledSplit(1, TABLE_SCAN_NODE_ID, DriverLifeCycle.taskWide(), 200000, 200300),
```

Comment:
I think this would be easier to understand if it was `begin` and `positions`, so we would see 300 here and could relate it to the value below